File: Java/DesignPatterns/src/main/java/designPatterns/creational/abstractFactory/ConcreteProduct.java
package designPatterns.creational.abstractFactory;

public class ConcreteProduct {
  static class ConcreteA implements AbstractProduct.A {
    @Override
    public String displayA() {
      return "Product A";
    }
  }

  static class ConcreteB implements AbstractProduct.B {
    @Override
    public String displayB() {
      return "Product B";
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/creational/abstractFactory/ConcreteProduct.kt
package creational.abstractFactory

sealed class ConcreteProduct {
  class ProductA : AbstractProduct.A {
    override fun displayA(): String {
      return "Product A"
    }
  }

  class ProductB : AbstractProduct.B {
    override fun displayB(): String {
      return "Product B"
    }
  }
}
